cpu/sam3x8e/periph/pwm.c

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright (C) 2015 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_sam3x8e
+ * @{
+ *
+ * @file
+ * @brief       CPU specific low-level PWM driver implementation for the SAM3X8E
+ *
+ * @author      Andreas "Paul" Pauli" <andreas.pauli@haw-hamburg.de>
+ *
+ * @}
+ */
+#include <stdint.h>
+#include "board.h"
+#include "periph/pwm.h"
+
+/*
+ * guard file in case no PWM device is defined,
+ */
+#if PWM_NUMOF && PWM_0_EN
+
+#define ERR_INIT_MODE           (-1)
+#define ERR_INIT_BWTH           (-2)
+#define ERR_SET_CHAN            (-1)
+
+/*
+ *
+ */
+int pwm_init(pwm_t dev, pwm_mode_t mode,
+             unsigned int frequency,
+             unsigned int resolution)
+{
+
+    int32_t retval = ERR_INIT_MODE;    /* Worst/First case */
+    uint32_t pwm_clk = 0;              /* Desired/real pwm_clock */
+    uint32_t diva = 1;                 /* Candidate for 8bit divider */
+    uint32_t prea = 0;                 /* Candidate for clock select */
+
+    switch (dev) {
+#if PWM_0_EN
```

Comment:
if ```#if PWM_0_EN``` shows up more than once in this file, the implementation is broken. (Don't copy the bad coding style of existing implementations.)